index.html

```diff
@@ -141,10 +232,40 @@
 
 
 
-	// ----------------- file selector-------------
+	// ------------------ gist publish --------------
+
+	$('#gist').click(function(){
+		if (confirm("Are you sure you want to publish all your files anonymously as a public gist on github.com")) {
+										   
+			var files = {};
+			var filesArr = getFiles();
+			var description = "Created using soleditor: Realtime Ethereum Contract Compiler and Runtime. Load this file by pasting this gists URL or ID at https://chriseth.github.io/browser-solidity#gist=";
+			
+			for(var f in filesArr) {
+				files[fileNameFromKey(filesArr[f])] = {
+					content: localStorage[filesArr[f]]
+				};
+			}
+
+			$.ajax({ 
+				url: 'https://api.github.com/gists',
+				type: 'POST',
+				data: JSON.stringify({
+					description: description,
+					public: true,
+					files: files
+				})
+			}).done(function(response) {
+				if (response.html_url && confirm("Created a gist at " + response.html_url + " Would you like to open it in a new window?")) {
+					window.open( response.html_url, '_blank' );
```

Comment:
@chriseth unfortunately the only way to prevent pop-blocker notification is to do the navigation on user click (confirm prompt doesn't count) I will look into a solution in a followup PR